### DISCUSSION TITLE
How do I setup presets?

### DISCUSSION THREAD

**Discussion by blockninja124** [misode/misode.github.io#628] [Q&A]
Last question, I promise :P

How do you add a preset to a generator? Like a pre-filled out example? Such as https://misode.github.io/worldgen/world-preset/?version=1.20&preset=debug_all_block_states 

All the modded generators seem to have no presets, is it possible?

Oh and is it possible to have a file be at a different namespace? Like to override my mods files, the generated files need to be under the cosmos namespace. Is there a way to force the files to save to that namespace?

**Comment by misode** [comment]
There's no easy way to do this currently. You'll have to modify code in two places in `services/DataFetcher.ts`:
* `fetchRegistries` is used to get a list of all preset IDs. After the for-loop you could do another fetch or a hardcoded list and add it to `result.set(<gen id>, [...])`
* `fetchPreset` is used to get the actual contents of a preset. There is already a hardcoded check for immersive weathering, so you could base it off that.

**Reply by blockninja124** [reply]
@misode So I've mostly managed to add to those functions, however while my presets now show up through changing the URL, they don't show up under the preset button. Is there perhaps another function I also need to change to have them show up there?

**Reply by blockninja124** [reply]
For now I'll try and finish up my generator and maybe start a PR. It'll work with presets if you change the URL and later I can try and figure out why they aren't in the preset menu
